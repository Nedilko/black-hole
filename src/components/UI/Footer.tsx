const Footer = () => {
  return (
    <div className="flex w-full text-3xl text-gray-300/80 uppercase justify-center border-t-2 pt-4 border-gray-400/80">
      <a
        className="hover:cursor-pointer hover:text-shadow hover:text-gray-300 transition-all duration-150"
        href="mailto:alexandr.nedilko@gmail.com"
      >
        contact me
      </a>
    </div>
  );
};

export default Footer;
