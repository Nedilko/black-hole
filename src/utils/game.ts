import { DIFFUCULTY, Results } from '../store/gameSlice';

export const getDefaultResults = (): Results => {
  return {
    [DIFFUCULTY.EASY]: 0,
    [DIFFUCULTY.MEDIUM]: 0,
    [DIFFUCULTY.HARD]: 0,
    [DIFFUCULTY.CUSTOM]: 0,
  };
};
