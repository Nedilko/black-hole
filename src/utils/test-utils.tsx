import { RenderOptions } from '@testing-library/react';
import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { PreloadedState } from 'redux';
import { AppStore, RootState, setupStore } from '../store';
import { render } from '@testing-library/react';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialState?: PreloadedState<RootState>;
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    initialState = {},
    store = setupStore(initialState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
};
