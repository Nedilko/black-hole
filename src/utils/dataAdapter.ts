import type { Results } from '../store/gameSlice';
import { getDefaultResults } from './game';
import {
  writeToLocalstorage,
  readFromLocalstorage,
} from './localstorageAdapter';

const loadResults = (): Results => {
  const localStorageData = readFromLocalstorage('results');
  if (!localStorageData) {
    writeResults(getDefaultResults());
    return getDefaultResults();
  }
  return JSON.parse(atob(localStorageData));
};

const writeResults = (settings: Results): void => {
  writeToLocalstorage('results', btoa(JSON.stringify(settings)));
};

export { loadResults, writeResults };
