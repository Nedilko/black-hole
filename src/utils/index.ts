import { getTime } from './time';
import {
  readFromLocalstorage,
  writeToLocalstorage,
} from './localstorageAdapter';
import { getDefaultResults } from './game';
import { loadResults, writeResults } from './dataAdapter';

export {
  getTime,
  readFromLocalstorage,
  writeToLocalstorage,
  getDefaultResults,
  loadResults,
  writeResults,
};
