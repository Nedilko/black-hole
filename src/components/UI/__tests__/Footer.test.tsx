import { renderWithProviders } from '../../../utils/test-utils';
import { screen } from '@testing-library/react';
import Footer from '../Footer';

describe('Footer', () => {
  it('should render Footer', () => {
    const { container } = renderWithProviders(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('should render exact text', () => {
    renderWithProviders(<Footer />);
    expect(screen.getByText(/contact me/)).toBeInTheDocument();
  });

  it('should correct mailto attribute', () => {
    renderWithProviders(<Footer />);
    expect(screen.getByText(/contact me/)).toHaveAttribute(
      'href',
      'mailto:alexandr.nedilko@gmail.com'
    );
  });
});
