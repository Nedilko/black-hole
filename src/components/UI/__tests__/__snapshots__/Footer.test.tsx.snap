// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render Footer 1`] = `
<div>
  <div
    class="flex w-full text-3xl text-gray-300/80 uppercase justify-center border-t-2 pt-4 border-gray-400/50"
  >
    <a
      class="hover:cursor-pointer hover:text-shadow hover:text-gray-300 transition-all duration-150"
      href="mailto:alexandr.nedilko@gmail.com"
    >
      contact me
    </a>
  </div>
</div>
`;
